lib/galaxy/workflow/errors.py

```diff
@@ -0,0 +1,207 @@
+        <tr style="background-color: #f2f2f2"><td>Last Update</td><td>${invocation_update_time}</td></tr>
+        <tr><td>Scheduling State</td><td>${invocation_scheduling_state}</td></tr>
+        <tr style="background-color: #f2f2f2"><td>Workflow ID</td><td>${workflow_id} (${workflow_id_encoded})</td></tr>
+        <tr><td>Workflow Version</td><td>${workflow_version}</td></tr>
+        <tr style="background-color: #f2f2f2"><td>Stored Workflow ID</td><td>${stored_workflow_id} (${stored_workflow_id_encoded})</td></tr>
+    </tbody>
+</table>
+
+This is an automated message. Do not reply to this address.
+</body></html>
+"""
+
+
+class WorkflowErrorReporter:
+    def __init__(self, invocation, app):
+        # Get the invocation
+        assert isinstance(invocation, model.WorkflowInvocation), ValueError(
+            f"Bad value provided for WorkflowInvocation ({invocation})."
+        )
+        self.invocation = invocation
+        self.workflow = invocation.workflow
+        self.history = invocation.history
+        self.app = app
+        self.report = None
+
+    def _can_access_invocation(self, trans, user):
+        if not user:
+            return False
+        if not trans:
+            return False
+        try:
+            self.app.workflow_manager.check_security(trans, self.invocation, check_ownership=False)
+            return True
+        except Exception:
+            return False
+
+    def create_report(self, user, email="", message="", redact_user_details_in_bugreport=False, **kwd):
+        host = self.app.url_for("/", qualified=True)
+
+        invocation_id_encoded = self.app.security.encode_id(self.invocation.id)
+        invocation_view_link = self.app.url_for(f"/workflows/invocations/{invocation_id_encoded}", qualified=True)
+        invocation_update_time = self.invocation.update_time.strftime("%d/%m/%Y %H:%M:%S")
+        invocation_scheduling_state = self.invocation.state
+
+        history_id_encoded = self.app.security.encode_id(self.history.id)
+        history_view_link = self.app.url_for("/histories/view", id=history_id_encoded, qualified=True)
+
+        stored_workflow_id_encoded = self.app.security.encode_id(self.workflow.stored_workflow.id)
+        workflow_id_encoded = self.app.security.encode_id(self.workflow.id)
+        workflow_version = self.workflow.version
+        workflow_view_link = self.app.url_for(
+            f"/published/workflow?id={stored_workflow_id_encoded}&version={workflow_version}", qualified=True
+        )
+
+        # TODO: We could maybe also include the invocation messages, but I believe they often link steps
+        # which aren't neccessarily the ones where the first jobs fail.
+        # invocation_messages = self.invocation.messages
+        # # Format messages for display
+        # if invocation_messages:
+        #     messages_str = "\n".join([f"- {msg[0].reason}" for msg in invocation_messages])
+        # else:
+        #     messages_str = "No messages"
```

Comment:
```suggestion
```
I wouldn't bother doing anything here, as an admin I think you'll have to look at the invocation anyway.